### ISSUE TITLE
How to calculate the probability of each pixel belongs to every class?

### ISSUE THREAD

**Issue by rita5022** [CSAILVision/sceneparsing#8]
After I ran the code,
I output the array "imPred" after line 66 in "demoSegmentation.m",
% imPred = net.forward({im_inp});
the imPred{1} array is 384 * 384 * 151,
so I expect to get the probability of each pixel belongs to each class,
for instance, 0.8, 0.53, 0.01...etc, which are between 0~1.

However the numbers I got from imPred{1} were like -1.2331, 3.0104, -0.7758, 10.1961...etc,
so I was wondering if these numbers can convert to probabilities,
and how to convert them?

Thank you.

**Comment by hangzhaomit** [comment]
Yes, you can simply calculate it by adding a softmax layer.

**Comment by rita5022** [comment]
Is it okay to fine-tuned the model from
http://sceneparsing.csail.mit.edu/model/FCN_iter_160000.caffemodel
http://sceneparsing.csail.mit.edu/model/DilatedNet_iter_120000.caffemodel
by adding a softmax layer to get the probabilities,
or it's better to run the caffe training code "ade_layers.py"?

By the way,
the file "train_DilatedNet.proto.txt" has already include a layer
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc_final_up"
  bottom: "label"
  top: "loss"
  loss_param {
    ignore_label: 0
  }
does it means that the model trained from this training architecture can output the probabilites?

Thank you!

**Comment by hangzhaomit** [comment]
Hi Rita, the softmax layer is not trainable, so no weights are included in the model file. Feel free to finetune our model. Thanks!
